### PR TITLE
Convert "shipment tracking providers" screen to MVVM

### LINKED ISSUE #2892: Refactor order detail sub screens to MVVM
The Order detail screen has been refactored to MVVM and the only sub screens that are pending a refactor is the
- Add Order note fragment
- Add shipment Tracking fragment

### PR DESCRIPTION
**Pull request by hichamboushaba**
The third and last part of #2892, it converts the screen `AddOrderTrackingProviderListFragment` to MVVM:

### Changes:
1. Converts AddOrderTrackingProviderListFragment to a regular fragment.
2. Update navigation to use navigation component.
3. Add an OrderShipmentProvidersRepository.
4. Converts the architecture to MVVM. 
5. Fixes an issue from my two previous PRs: cleaning up the repository to avoid leaking the eventbus subscription 1a82d753cb7ab4d5f4a7b2034c690137ee899f89

### Testing:
Test different scenarios of the AddOrderTrackingProviderListFragment.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/AddOrderTrackingProviderListViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+class AddOrderTrackingProviderListViewModel @AssistedInject constructor(
+    @Assisted savedState: SavedStateWithArgs,
+    dispatchers: CoroutineDispatchers,
+    private val shipmentProvidersRepository: OrderShipmentProvidersRepository,
+    private val orderDetailRepository: OrderDetailRepository,
+    private val resourceProvider: ResourceProvider
+) : ScopedViewModel(savedState, dispatchers) {
+    private val navArgs: AddOrderTrackingProviderListFragmentArgs by savedState.navArgs()
+
+    val TrackingProviderListViewStateData = LiveDataDelegate(
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/AddOrderTrackingProviderListViewModel.kt]: Nit-pick: Normally we use lower-case letters for properties. I checked that the upper-case is used if it's a reference to an `object`.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/AddOrderTrackingProviderListFragment.kt [hunk 2]
```diff
@@ -149,76 +78,67 @@ class AddOrderTrackingProviderListFragment : DialogFragment(), AddOrderTrackingP
...
-
         super.onDestroyView()
     }
 
-    override fun getCountryName(): String? {
-        context?.let {
-            return StringUtils.getCountryByCountryCode(it, presenter.loadStoreCountryFromDb())
+    private fun setupObservers(binding: DialogOrderTrackingProviderListBinding) {
+        viewModel.TrackingProviderListViewStateData.observe(viewLifecycleOwner) { old, new ->
+            new.providersList.takeIfNotEqualTo(old?.providersList) {
```
Comment (0nko) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/AddOrderTrackingProviderListFragment.kt:97]: The `query` observer is missing, so the search doesn't work.

**Review by 0nko — CHANGES_REQUESTED** [review]
The changes look good 👍, but the search doesn't work (more details in the code).

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/AddOrderTrackingProviderListFragment.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/AddOrderTrackingProviderListFragment.kt:97]: I can't reproduce the issue of search, it's working for me, but I'll try to check it more, now that I think about it, having the same menu inflated here and in OrdersListFragment may cause an issue, as `onCreateOptionsMenu` will be called for both fragments (the same thing that caused the crash we just fixed)

But I don't think that the `query` has to be observed here, its value is just to restore the query text, which is done in `onCreateOptionsMenu`, and the filtering and propagating the filtered list is done here: https://github.com/woocommerce/woocommerce-android/blob/3ca9e474b7fe330981595059d1dcc2c330213fe0/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/AddOrderTrackingProviderListViewModel.kt#L78-L90

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/AddOrderTrackingProviderListViewModel.kt (hunk 1 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/AddOrderTrackingProviderListViewModel.kt]: Good catch, autocomplete is useful, but sometimes it does thing like this 😄, I'll fix it 👍

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/AddOrderTrackingProviderListFragment.kt (hunk 2 above)
Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/tracking/AddOrderTrackingProviderListFragment.kt:97]: @0nko I was able to confirm the issue I described above after screen rotation, I added the commit 157c6b848e06c58cb4237f549ac8b4fba4d30f4d to use a separate menu with a separate id for the search view to fix it.

**Comment by 0nko** [comment]
@hichamboushaba The search works now 👍. Just `TrackingProviderListViewStateData` needs to be changed and we can merge it.

**Comment by hichamboushaba** [comment]
> @hichamboushaba The search works now 👍. Just `TrackingProviderListViewStateData` needs to be changed and we can merge it.

@0nko done eea1556, I forgot about it 🤦

**Review by 0nko — APPROVED** [review]
Nice work! :shipit:

**Pull request merged**